### ISSUE TITLE
ReorientTetMesh() in two-level method with H(div) and H(curl) spaces

### ISSUE THREAD

**Issue by kvoronin** [mfem/mfem#424]
Hi all,

I'd like to have the two-level setup where both RT for H(div) and ND for H(curl) spaces are used (3D case).
So, when there was no H(div), I created a ParMesh pmesh, and two more ParMesh objects pmesh_lvls[0-1] (all pointers), so I did something like
```
// having a pmesh defined
Hdiv_space = new ParFiniteElementSpace(pmesh, hdiv_coll); 
...
pmesh_lvls[0] = new ParMesh(*pmesh)
// define the f.e. space for the coarse mesh
pmesh->UniformRefiment();
pmesh_lvls[1] = new ParMesh(*pmesh);
// define f.e. space for the fine mesh
// and call Hdiv_space->Update, GetUpdateOperator, Dof_TrueDof_Matrix() to create a global interpolation matrix between levels, e.g.
```
But when I have also H(curl) in the setup, I have to call ReorientTetMesh for the pmesh in order to define the Nedelec space which will be later called with Update() after refinement. But I cannot call UniformRefinement() after ReorientTetMesh(), I got errors like 
```
MFEM abort: (r,c,f) = (100,289,617)
 ... in function: int mfem::STable3D::operator()(int, int, int) const
 ... at line 83 of file: general/stable3d.cpp
```

How can I avoid that?

Thanks,
Kirill

**Comment by tzanio** [comment]
This is a known limitation, see

https://github.com/mfem/mfem/blob/115474df1ead1e9feffe3a023b375a64a9092b79/mesh/mesh.hpp#L883-L891

In your particular case, you may be able to get away with using two separate mesh objects -- one coarse and one refined, but we don't have a good solution for this in general.

Tzanio

**Comment by kvoronin** [comment]
Thanks, @tzanio!
I am afraid I cannot get away, because I need to have a f.e. space (Nedelec) defined on the mesh which gets refined, so that I can call Update() and f.e. space would know that the mesh has been refined and would give me the Update operator.

**Comment by tzanio** [comment]
Can you run on hex meshes? This limitation is just for high-order H(curl) on tets :)

**Comment by kvoronin** [comment]
Hex meshes don't fit well, but I wonder what is the high-order? I am fine with the first-order Nedelec elements, still see the errors.

**Comment by chikitkin** [comment]
> This is a known limitation, see
> 
> https://github.com/mfem/mfem/blob/115474df1ead1e9feffe3a023b375a64a9092b79/mesh/mesh.hpp#L883-L891
> 
> In your particular case, you may be able to get away with using two separate mesh objects -- one coarse and one refined, but we don't have a good solution for this in general.
> 
> Tzanio

Hi, Tzanio,
I faced with the same problem. Could you, please, explain what do you mean by "using two separate mesh objects"? Should I create another ```pmesh``` object from the Reoriented mesh? But how it actually helps?

Thank you in advance!
Aleksandr Chikitkin

**Comment by tzanio** [comment]
Hi Aleksandr,

This was actually just addressed by @mlstowell in PR https://github.com/mfem/mfem/pull/1046.

Can you try this branch and let us know if it is working for you?

Tzanio

**Comment by chikitkin** [comment]
Thanks! I will try it.

**Comment by chikitkin** [comment]
> Hi Aleksandr,
> 
> This was actually just addressed by @mlstowell in PR #1046.
> 
> Can you try this branch and let us know if it is working for you?
> 
> Tzanio

I checked, it works, thank you!
